distribution/helm/vector/templates/rbac.yaml

```diff
@@ -0,0 +1,93 @@
+{{- if .Values.rbac.enabled -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "vector.serviceAccountName" . }}
+  labels:
+    {{- include "vector.labels" . | nindent 4 }}
+  {{- with .Values.rbac.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+---
+apiVersion: {{ default (include "rbac.apiVersion" .) .Values.rbac.apiVersion }}
+kind: ClusterRole
+metadata:
+  name: {{ template "vector.fullname" . }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - get
+{{- if .Values.rbac.psp.enabled }}
+  - apiGroups:
+      - policy
+    resources:
+      - podsecuritypolicies
+    resourceNames:
+      - {{ template "vector.fullname" . }}
+    verbs:
+      - use
+{{- end }}
+---
+apiVersion: {{ default (include "rbac.apiVersion" .) .Values.rbac.apiVersion }}
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "vector.fullname" . }}
+  labels:
+    {{- include "vector.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "vector.fullname" . }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ include "vector.serviceAccountName" . }}
+    namespace: {{ .Release.Namespace }}
+---
+{{- if .Values.rbac.psp.enabled }}
+apiVersion: {{ default (include "psp.apiVersion" .) .Values.rbac.psp.apiVersion }}
+kind: PodSecurityPolicy
```

Comment:
Minor nit: `PodSecurityPolicy` isn't really a part of RBAC, it should probably be in a separate file.